gtest_parallel.py

```diff
@@ -358,6 +364,42 @@ def dump_to_file_and_close(self):
 
 # Record of test runtimes. Has built-in locking.
 class TestTimes(object):
+  class LockedFile(object):
+    def __init__(self, filename, mode):
+      self._filename = filename
+      self._mode = mode
+      self._fo = None
+
+    def __enter__(self):
+      self._fo = open(self._filename, self._mode)
+
+      try:
+        if sys.platform == 'win32':
+          msvcrt.locking(self._fo.fileno(), msvcrt.LK_LOCK, 1)
```

Comment:
> Also, are we sure that _fo is always at position 0 here?

Good point. I'll add `self._fo.seek(0)` in `__enter__` and `__exit__` on win32.

> Also, are any of these methods reentrant, or do they both fail if a thread locks a segment that another thread has locked?

If I understood you correctly then both of them block until other thread releases the lock.